### PR TITLE
Upgrade pdal version to 2.3.0

### PR DESCRIPTION
**Pull request by loic-lopez**
Specify library name and version:  **pdal/2.3.0**

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

Resolves: #7532

Tests done:

- `conan create conanfile.py pdal/2.2.0@`
- `conan create conanfile.py pdal/2.3.0@`

### CODE REVIEW

**Comment by SSE4** [comment]
> It's a good question, I'm wondering why this issue appears only on `macOS, Clang (M1/arm64)`

it's the ARM build, which is actually a cross-compiler. perhaps, this executable was built for ARM instead of x86, so it cannot be launched.
if cross-building is not supported for pdal right now, you may raise `ConanInvalidConfiguration`.

**Comment by loic-lopez** [comment]
> Indeed, there were some missing packages: https://c3i.jfrog.io/c3i/misc/summary.html?json=https://c3i.jfrog.io/c3i/misc/logs/TapaholesList/252/summary.json

Thanks for the help, have you got an idea why INVALID_CONFIGURATION is thrown with x86_64 architecture on windows?

File: recipes/pdal/all/conandata.yml [hunk 1]
```diff
@@ -16,3 +19,8 @@ patches:
       base_path: "source_subfolder"
     - patch_file: "patches/0006-pointview.patch"
       base_path: "source_subfolder"
+  "2.3.0":
+    - patch_file: "patches/0001-cmakelists.patch"
+      base_path: "source_subfolder"
+    - patch_file: "patches/0005-vendor-nanoflann.patch"
+      base_path: "source_subfolder"
```
Comment (SpaceIm) [inline recipes/pdal/all/conandata.yml:26]: what about cmake-deps patch? Is it not needed in 2.3.0 for a robust build?

File: recipes/pdal/all/conanfile.py [hunk 2]
```diff
@@ -92,6 +92,8 @@ def validate(self):
         miss_boost_required_comp = any(getattr(self.options["boost"], "without_{}".format(boost_comp), True) for boost_comp in self._required_boost_components)
         if self.options["boost"].header_only or miss_boost_required_comp:
             raise ConanInvalidConfiguration("{0} requires non header-only boost with these components: {1}".format(self.name, ", ".join(self._required_boost_components)))
+        if self.settings.arch not in ("x86_64", ):
+            raise ConanInvalidConfiguration(f"Unsupported arch={self.settings.arch}")
```
Comment (SpaceIm) [inline recipes/pdal/all/conanfile.py:96]: It doesn't support others arch or it just cross-building?

**Pull request merged**